src/content/components/TreeView.js

```diff
@@ -119,16 +120,25 @@ class TreeViewRowScrolledColumns extends Component {
     }
   }
 
+  _renderIcon(iconUrl) {
+    return <img
+            src={iconUrl}
+            referrerPolicy='no-referrer'
+            className='treeRowIcon'
+            onError={ e => this.props.onIconError(this.props.nodeId, e.target.src) } />;
+  }
+
   render() {
     const {
       node, depth, mainColumn, appendageColumn, index, canBeExpanded,
-      isExpanded, selected, highlightString, appendageButtons,
+      isExpanded, selected, highlightString, appendageButtons, icon,
     } = this.props;
     const evenOddClassName = (index % 2) === 0 ? 'even' : 'odd';
     return (
       <div className={`treeViewRow treeViewRowScrolledColumns ${evenOddClassName} ${selected ? 'selected' : ''} ${node.dim ? 'dim' : ''}`} style={{height: '16px'}} onClick={this._onClick}>
         <span className='treeRowIndentSpacer' style={{ width: `${depth * 10}px` }}/>
         <span className={`treeRowToggleButton ${isExpanded ? 'expanded' : 'collapsed'} ${canBeExpanded ? 'canBeExpanded' : 'leaf'}`} />
+        { icon && this._renderIcon(icon) }
```

Comment:
got it :)

I used it because React's doc give it as an example (https://facebook.github.io/react/docs/conditional-rendering.html#inline-if-with-logical--operator and https://facebook.github.io/react/docs/jsx-in-depth.html#booleans-null-and-undefined-are-ignored). But really I have no strong opinion so I'll do as you suggest :)